packages/portal-proto/src/components/Modals/SaveCohortModal.tsx

```diff
@@ -46,23 +51,59 @@ const SaveCohortModal = ({
     await addCohort({ cohort: addBody, delete_existing: replace })
       .unwrap()
       .then((payload) => {
-        coreDispatch(copyCohort({ sourceId: prevCohort, destId: payload.id }));
-        // NOTE: the current cohort can not be undefined. Setting the id to a cohort
-        // which does not exist will cause this
-        // Therefore, copy the unsaved cohort to the new cohort id received from
-        // the BE.
-        coreDispatch(
-          setCohortMessage([`savedCohort|${newName}|${payload.id}`]),
-        );
-        coreDispatch(
-          removeCohort({
-            shouldShowMessage: false,
-            currentID: prevCohort,
-          }),
-        );
-        coreDispatch(setCurrentCohortId(payload.id));
-        coreDispatch(updateCohortName(newName));
-        coreDispatch(fetchCohortCaseCounts(payload.id));
+        if (prevCohort) {
+          coreDispatch(
+            copyCohort({
+              sourceId: prevCohort,
+              destId: payload.id,
+              saved: true,
+            }),
+          );
+          // NOTE: the current cohort can not be undefined. Setting the id to a cohort
+          // which does not exist will cause this
+          // Therefore, copy the unsaved cohort to the new cohort id received from
+          // the BE.
+          coreDispatch(
+            setCohortMessage([`savedCurrentCohort|${newName}|${payload.id}`]),
+          );
+          coreDispatch(
+            removeCohort({
+              shouldShowMessage: false,
+              currentID: prevCohort,
+            }),
+          );
+          coreDispatch(setCurrentCohortId(payload.id));
+          coreDispatch(updateCohortName(newName));
+          coreDispatch(fetchCohortCaseCounts(payload.id));
+        } else {
+          coreDispatch(
+            setCohort({
+              id: payload.id,
+              name: payload.name,
+              filters: buildGqlOperationToFilterSet(payload.filters),
+              caseSet: { status: "uninitialized" },
+              counts: {
+                ...NullCountsData,
+              },
+              modified_datetime: payload.modified_datetime,
+              saved: true,
+              modified: false,
+            }),
+          );
+          if (setAsCurrent) {
+            coreDispatch(setCurrentCohortId(payload.id));
+            coreDispatch(
+              setCohortMessage([`savedCohort|${newName}|${payload.id}`]),
+            );
+          } else {
+            coreDispatch(
+              setCohortMessage([
+                `savedCohortSetCurrent|${payload.name}|${payload.id}`,
+              ]),
+            );
+          }
```

Comment:
Need to get the caseCount here as well, so add:
`coreDispatch(fetchCohortCaseCounts(payload.id));`

This will fix the spinner issue.